### ISSUE TITLE
[distributed] mc ls command throws XMinioServerNotInitialized

### ISSUE THREAD

**Issue by balamurugana** [minio/minio#2508]
I have used two machines `m2.minio.io` and `m3.minio.io` and from each machine `/mnt/sdk1/bala/mydisk` and `/mnt/sdi1/bala/mydisk` are used.

Minio Version:

``` bash
Version: 2016-08-20T07:51:29Z
Release-Tag: DEVELOPMENT.2016-08-20T07-51-29Z
Commit-ID: 38386510a41eca468e0bb05af8f092f94a6c97ad
```

~/.minio/config.json on all servers:

```
{
        "version": "6",
        "credential": {
                "accessKey": "minio",
                "secretKey": "minio123"
        },
        "region": "us-east-1",
        "logger": {
                "console": {
                        "enable": true,
                        "level": "debug"
                },
                "file": {
                        "enable": false,
                        "fileName": "",
                        "level": ""
                },
                "syslog": {
                        "enable": false,
                        "address": "",
                        "level": ""
                }
        },
        "notify": {
                "amqp": null,
                "elasticsearch": null,
                "redis": null
        }
}
```

Minio started on all servers:

``` bash
minio server --address=:60000 m2.minio.io:/mnt/sdk1/bala/mydisk m3.minio.io:/mnt/sdk1/bala/mydisk  m2.minio.io:/mnt/sdi1/bala/mydisk m3.minio.io:/mnt/sdi1/bala/mydisk
```

mc command:

``` bash
$ mc --debug ls myminio/

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T104942Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:49:42 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 4A4ELIXCL4BG133L

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  301.233271ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T104942Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:49:42 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: MCFPZCF7M4DMSOCC

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  303.639063ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T104944Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:49:44 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: BFG19LWBLEY69R0Z

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  282.367513ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T104947Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:49:47 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: R00MNFJBKNVOQV3I

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  306.356208ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T104947Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:49:47 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: WSI0AFVPKW4ICJ59

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  277.909235ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T105002Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:50:03 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: EDZCO468GN7SH697

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  282.302026ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T105003Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:50:03 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: W4AGZS2K3NVDC06R

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  299.921013ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T105005Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:50:05 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: QH0Y633A48JZ49CK

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  281.083092ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T105008Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:50:08 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: L9ECQ7J7CBB61T0E

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  281.285105ms

mc: <DEBUG> GET / HTTP/1.1
Host: 65.19.167.92:60000
User-Agent: Minio (linux; amd64) minio-go/2.0.1 mc/2016-08-20T10:30:20Z
Authorization: AWS4-HMAC-SHA256 Credential=minio/20160820/us-east-1/s3/aws4_request, SignedHeaders=expect;host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
Expect: 100-continue
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20160820T105010Z
Accept-Encoding: gzip

mc: <DEBUG> HTTP/1.1 503 Service Unavailable
Transfer-Encoding: chunked
Accept-Ranges: bytes
Content-Type: text/xml; charset=utf-8
Date: Sat, 20 Aug 2016 10:50:10 GMT
Server: Minio/DEVELOPMENT.2016-08-20T07-51-29Z (linux; amd64)
Vary: Origin
X-Amz-Request-Id: 1XQZC3R5IHUJMO40

105
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><Key></Key><BucketName></BucketName><Resource>/</Resource><RequestId>3L137</RequestId><HostId>3L137</HostId></Error>
0

mc: <DEBUG> Response Time:  300.837791ms

mc: <ERROR> Unable to list folder. 503 Service Unavailable
 (1) ls.go:133 cmd.doList(..) Tags: [http://65.19.167.92:60000/]
 (0) client-s3.go:975 cmd.(*s3Client).listInRoutine(..)
 Release-Tag:DEVELOPMENT.2016-08-20T10-30-20Z | Commit:a885ffd64bc3 | Host:minio2 | OS:linux | Arch:amd64 | Lang:go1.6.3 | Mem:2.6MB/6.6MB | Heap:2.6MB/3.6MB
```

**Comment by balamurugana** [comment]
When minio was compiled using `go build` command it works whereas using `make` fails with the above behaviour.

**Comment by harshavardhana** [comment]
Fixed by 81da694efccd02231d50991146f64d553ad1fc31
